tgui/packages/tgui/interfaces/NtosCameraConsole.js

```diff
@@ -0,0 +1,94 @@
+import { filter, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { Fragment } from 'inferno';
+import { useBackend, useLocalState } from '../backend';
+import { Button, ByondUi, Input, Section } from '../components';
+import { NtosWindow } from '../layouts';
+import { CameraConsoleContent } from './CameraConsole';
+import { logger } from "../logging";
+
+/**
+ * Returns previous and next camera names relative to the currently
+ * active camera.
+ */
+const prevNextCamera = (cameras, activeCamera) => {
+  if (!activeCamera) {
+    return [];
+  }
+  const index = cameras.findIndex(camera => (
+    camera.name === activeCamera.name
+  ));
+  return [
+    cameras[index - 1]?.name,
+    cameras[index + 1]?.name,
+  ];
+};
+
+/**
+ * Camera selector.
+ *
+ * Filters cameras, applies search terms and sorts the alphabetically.
+ */
+const selectCameras = (cameras, searchText = '') => {
+  const testSearch = createSearch(searchText, camera => camera.name);
+  return flow([
+    // Null camera filter
+    filter(camera => camera?.name),
+    // Optional search term
+    searchText && filter(testSearch),
+    // Slightly expensive, but way better than sorting in BYOND
+    sortBy(camera => camera.name),
+  ])(cameras);
+};
+
+export const NtosCameraConsole = (props, context) => {
+  const { act, data, config } = useBackend(context);
+  const { PC_device_theme, mapRef, activeCamera } = data;
+  const cameras = selectCameras(data.cameras);
+  const [
+    prevCameraName,
+    nextCameraName,
+  ] = prevNextCamera(cameras, activeCamera);
+  return (
+    <NtosWindow
+      width={800}
+      height={600}
+      theme={PC_device_theme}>
+      <div className="CameraConsole__left">
+        <NtosWindow.Content scrollable>
+          <CameraConsoleContent />
+        </NtosWindow.Content>
+      </div>
+      <div className="CameraConsole__right">
+        <div className="CameraConsole__toolbar">
+          <b>Camera: </b>
+          {activeCamera
+            && activeCamera.name
+            || '—'}
+        </div>
+        <div className="CameraConsole__toolbarRight">
+          <Button
+            icon="chevron-left"
+            disabled={!prevCameraName}
+            onClick={() => act('switch_camera', {
+              name: prevCameraName,
+            })} />
+          <Button
+            icon="chevron-right"
+            disabled={!nextCameraName}
+            onClick={() => act('switch_camera', {
+              name: nextCameraName,
+            })} />
+        </div>
+        <ByondUi
+          className="CameraConsole__map"
+          params={{
+            id: mapRef,
+            type: 'map',
+          }} />
+      </div>
+    </NtosWindow>
+  );
+};
```

Comment:
```suggestion
};

```
Travis hungers for newlines

Reply:
@zxaber install editorconfig extension for vscode, it automatically does this.